lib/cancan/model_adapters/active_record_4_adapter.rb

```diff
@@ -34,10 +34,15 @@ def matches_condition?(subject, name, value)
       # look inside the where clause to decide to outer join tables
       # you're using in the where. Instead, `references()` is required
       # in addition to `includes()` to force the outer join.
-      def build_relation(*where_conditions)
-        relation = @model_class.where(*where_conditions)
-        relation = relation.includes(joins).references(joins) if joins.present?
-        relation
+      def build_joins_relation(relation, *_where_conditions)
+        case CanCan.accessible_by_strategy
+        when :subquery
+          # subquery mode doesn't work with Rails 4.x
+          relation.includes(joins).references(joins)
+
+        when :left_join
+          relation.includes(joins).references(joins)
+        end
```

Comment:
I guess I am missing something here 😅  why does it do the same?

Reply:
I'll tidy this up a bit. The config option really only works with AR5+.